unstructured/ingest/main.py

```diff
@@ -392,6 +395,12 @@ def run(self):
     show_default=True,
     help="Number of parallel processes to process docs in.",
 )
+@click.option(
+    "--recursive",
+    is_flag=True,
+    default=False,
+    help="Will recurse directories for S3, GCS and Azure storage.",
```

Comment:
Obviously bump the types included if we go for the suggestion above. Maybe slight tweak to follow description pattern like remote-url to something like:
```suggestion
    help="Recursively download files in their respective folders"
    "otherwise stop at the files in provided folder level."
    " Supported protocols are: `gcs`, `gs`, `s3`, `s3a`, `abfs` "
    "and `az`.",
```

Reply:
Done.